core/src/main/java/jenkins/util/JSONSignatureValidator.java

```diff
@@ -82,7 +82,11 @@ public FormValidation verifySignature(JSONObject o) throws IOException {
 
             // this is for computing a signature
             Signature sig = Signature.getInstance("SHA1withRSA");
-            sig.initVerify(certs.get(0));
+            if (certs.isEmpty()) {
+                return FormValidation.error("No certificate found in " + name + ". Cannot verify the signature");
```

Comment:
use `FormValidation.error("No certificate found in %s. Cannot verify the signature", name)` style as more readable